linux/switchtec.h

```diff
@@ -34,15 +34,16 @@
 #define SWITCHTEC_EVENT_FATAL    BIT(4)
 
 #define SWITCHTEC_DMA_MRPC_EN	BIT(0)
+
+#define SWITCHTEC_GAS_MRPC_SIZE	4096
 enum {
-	SWITCHTEC_GAS_MRPC_OFFSET       = 0x0000,
-	SWITCHTEC_GAS_TOP_CFG_OFFSET    = 0x1000,
-	SWITCHTEC_GAS_SW_EVENT_OFFSET   = 0x1800,
-	SWITCHTEC_GAS_SYS_INFO_OFFSET   = 0x2000,
-	SWITCHTEC_GAS_FLASH_INFO_OFFSET = 0x2200,
-	SWITCHTEC_GAS_PART_CFG_OFFSET   = 0x4000,
-	SWITCHTEC_GAS_NTB_OFFSET        = 0x10000,
-	SWITCHTEC_GAS_PFF_CSR_OFFSET    = 0x134000,
+	SWITCHTEC_GAS_TOP_CFG_OFFSET    = 0x0000,
+	SWITCHTEC_GAS_SW_EVENT_OFFSET   = 0x0800,
+	SWITCHTEC_GAS_SYS_INFO_OFFSET   = 0x1000,
+	SWITCHTEC_GAS_FLASH_INFO_OFFSET = 0x1200,
+	SWITCHTEC_GAS_PART_CFG_OFFSET   = 0x3000,
+	SWITCHTEC_GAS_NTB_OFFSET        = 0xf000,
+	SWITCHTEC_GAS_PFF_CSR_OFFSET    = 0x133000,
```

Comment:
I'd prefer it if you did this change without modifying these defines. They describe the hardware not how the software uses the hardware. And it would just be confusing if they are wrong.

So just subtract SWITCHTEC_GAS_TOP_CFG_OFFSET from any mapping that's done with the offset.
Therefore you only need to change anything that uses 'stdev->mmio' which isn't very much and is easy to grep for.